From 5aa97fa172e6c3bb0d269d67e061fdfe757a22f7 Mon Sep 17 00:00:00 2001
From: Joe Maples <joe@frap129.org>
Date: Tue, 28 Aug 2018 08:56:30 -0400
Subject: [PATCH 10/19] core: Make LOCAL_SDCLANG_2 no-op

AOSP Clang performs better than SDClang versions prior to 6.0, so
make LOCAL_SDCLANG_2 no-op to use AOSP Clang.

Change-Id: Iad65f5fae8a07a843aac7e42b4aef2a83da20a50
Signed-off-by: Joe Maples <joe@frap129.org>
---
 core/binary.mk         | 15 ---------------
 core/definitions.mk    |  3 ---
 core/executable.mk     |  6 ------
 core/shared_library.mk |  6 ------
 core/static_library.mk |  6 ------
 5 files changed, 36 deletions(-)

diff --git a/core/binary.mk b/core/binary.mk
index bdebda5bc..5ad863e2e 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -390,13 +390,6 @@ ifeq ($(my_sdclang),true)
         $(error LOCAL_SDCLANG and LOCAL_SDCLANG_2 can not be set to true at the same time!)
     endif
 endif
-ifeq ($(SDCLANG),true)
-    ifeq ($(my_sdclang),)
-        ifneq ($(my_sdclang_2),true)
-            my_sdclang := true
-        endif
-    endif
-endif
 
 ifeq ($(LOCAL_C_STD),)
     my_c_std_version := $(DEFAULT_C_STD_VERSION)
@@ -566,14 +559,6 @@ my_target_global_ldflags := $($(LOCAL_2ND_ARCH_VAR_PREFIX)CLANG_$(my_prefix)GLOB
             my_cxx := $(SDCLANG_PATH)/clang++
         endif
     endif
-    ifeq ($(my_sdclang_2),true)
-        ifeq ($(strip $(my_cc)),)
-            my_cc := $(SDCLANG_PATH_2)/clang
-        endif
-        ifeq ($(strip $(my_cxx)),)
-            my_cxx := $(SDCLANG_PATH_2)/clang++
-        endif
-    endif
 else
 my_target_global_cflags := $($(LOCAL_2ND_ARCH_VAR_PREFIX)$(my_prefix)GLOBAL_CFLAGS)
 my_target_global_conlyflags := $($(LOCAL_2ND_ARCH_VAR_PREFIX)$(my_prefix)GLOBAL_CONLYFLAGS) $(my_c_std_conlyflags)
diff --git a/core/definitions.mk b/core/definitions.mk
index 8c3405ddd..971c2139c 100644
--- a/core/definitions.mk
+++ b/core/definitions.mk
@@ -1256,7 +1256,6 @@ $(if $(PRIVATE_TIDY_CHECKS),$(clang-tidy-cpp))
 $(hide) $(RELATIVE_PWD) $(PRIVATE_CXX) \
   $(transform-cpp-to-o-compiler-args) \
   $(if $(findstring $(SDCLANG_PATH),$(PRIVATE_CXX)),$(SDCLANG_COMMON_FLAGS)) \
-  $(if $(findstring $(SDCLANG_PATH_2),$(PRIVATE_CXX)),$(SDCLANG_COMMON_FLAGS_2)) \
   -MD -MF $(patsubst %.o,%.d,$@) -o $@ $<
 endef
 endif
@@ -1306,7 +1305,6 @@ $(if $(PRIVATE_TIDY_CHECKS),$(clang-tidy-c))
 $(hide) $(RELATIVE_PWD) $(PRIVATE_CC) \
   $(transform-c-to-o-compiler-args) \
   $(if $(findstring $(SDCLANG_PATH),$(PRIVATE_CC)),$(SDCLANG_COMMON_FLAGS)) \
-  $(if $(findstring $(SDCLANG_PATH_2),$(PRIVATE_CC)),$(SDCLANG_COMMON_FLAGS_2)) \
   -MD -MF $(patsubst %.o,%.d,$@) -o $@ $<
 endef
 endif
@@ -1317,7 +1315,6 @@ define transform-s-to-o
 $(RELATIVE_PWD) $(PRIVATE_CC) \
   $(call transform-c-or-s-to-o-compiler-args, $(PRIVATE_ASFLAGS)) \
   $(if $(findstring $(SDCLANG_PATH),$(PRIVATE_CC)),$(SDCLANG_COMMON_FLAGS)) \
-  $(if $(findstring $(SDCLANG_PATH_2),$(PRIVATE_CC)),$(SDCLANG_COMMON_FLAGS_2)) \
   -MD -MF $(patsubst %.o,%.d,$@) -o $@ $<
 endef
 
diff --git a/core/executable.mk b/core/executable.mk
index b4bd3bacb..457a9e2a0 100644
--- a/core/executable.mk
+++ b/core/executable.mk
@@ -98,10 +98,4 @@ include $(SDCLANG_LTO_DEFS)
 endif
 endif
 
-ifeq ($(LOCAL_SDCLANG_2),true)
-ifeq ($(LOCAL_SDCLANG_LTO),true)
-include $(SDCLANG_LTO_DEFS)
-endif
-endif
-
 endif
diff --git a/core/shared_library.mk b/core/shared_library.mk
index eaaca8ea6..4f4b9e7b4 100644
--- a/core/shared_library.mk
+++ b/core/shared_library.mk
@@ -54,12 +54,6 @@ include $(SDCLANG_LTO_DEFS)
 endif
 endif
 
-ifeq ($(LOCAL_SDCLANG_2),true)
-ifeq ($(LOCAL_SDCLANG_LTO),true)
-include $(SDCLANG_LTO_DEFS)
-endif
-endif
-
 my_module_arch_supported :=
 
 ###########################################################
diff --git a/core/static_library.mk b/core/static_library.mk
index 7daa04143..a6d1a065e 100644
--- a/core/static_library.mk
+++ b/core/static_library.mk
@@ -40,12 +40,6 @@ include $(SDCLANG_LTO_DEFS)
 endif
 endif
 
-ifeq ($(LOCAL_SDCLANG_2),true)
-ifeq ($(LOCAL_SDCLANG_LTO),true)
-include $(SDCLANG_LTO_DEFS)
-endif
-endif
-
 my_module_arch_supported :=
 
 ###########################################################
-- 
2.19.1

